src/proof/mod.rs

```diff
@@ -110,16 +110,17 @@ pub trait Prover<'a, 'b, F: LurkField, C: Coprocessor<F>> {
         &self,
         multiframes: &'a [MultiFrame<'_, F, IO<F>, Witness<F>, C>],
     ) -> Result<SequentialCS<'a, F, IO<F>, Witness<F>, C>, SynthesisError> {
-        let res = multiframes
+        // Note: This loop terminates and throws an error on the first occurrence of `SynthesisError`.
+        multiframes
             .iter()
-            .enumerate()
-            .map(|(_, multiframe)| {
+            .map(|multiframe| {
                 let mut cs = TestConstraintSystem::new();
 
-                multiframe.clone().synthesize(&mut cs).unwrap(); // FIXME: unwrap
-                (multiframe.clone(), cs)
+                multiframe
+                    .clone()
+                    .synthesize(&mut cs)
+                    .map(|_| (multiframe.clone(), cs))
```

Comment:
Can you figure out how to avoid over-cloning `multiframe`? The signature of `synthesize` does not make it obvious why it should be the case.

Reply:
Despite my best efforts wrangling with the borrow checker, I wasn't able to figure out how to get around the cloning... I'm curious what you have in mind, so I can learn my way around these kinds of situations.

Reply:
Actually, my apologies, I read the wrong definition of the wrong synthesis method. It does take self by value, this is fine.